### PR TITLE
DR-1376: Apply semantic versioning in GCR on dev merge

### PR DESCRIPTION
**Pull request by fboulnois**
Currently, we update the semantic version of the repository and the repository tag on every dev merge. However, the containers that are pushed into GCR do not have the associated semantic version (they have a git hash). This means that if there were many dev updates in a day then only a single container per night is appropriately tagged with a semantic version in GCR, and there are "gaps" between versions which causes issues. This updates the actions so that the semantic versioning in the repo is in sync with the semantic versioning in GCR.

### CODE REVIEW

File: .github/workflows/dev-image-update.yaml
Comment (fboulnois) [inline .github/workflows/dev-image-update.yaml]: Needs to be forked before merging.

**Review by snf2ye — APPROVED** [review]
LGTM - good to merge after forking benc-uk/workflow-dispatch action

**Review by smark88 — APPROVED** [review]

**Pull request merged**
